frontend/src/modules/Shares/components/ShareBoxList.js

```diff
@@ -0,0 +1,489 @@
+import {
+  Autocomplete,
+  Box,
+  Checkbox,
+  Container,
+  Grid,
+  TextField,
+  Typography
+} from '@mui/material';
+import CircularProgress from '@mui/material/CircularProgress';
+import CheckBoxOutlineBlankIcon from '@mui/icons-material/CheckBoxOutlineBlank';
+import CheckBoxIcon from '@mui/icons-material/CheckBox';
+import PropTypes from 'prop-types';
+import { useCallback, useEffect, useState } from 'react';
+import { Helmet } from 'react-helmet-async';
+import { Defaults, Pager, ShareStatus, useSettings } from 'design';
+import { SET_ERROR, useDispatch } from 'globalErrors';
+import {
+  listDatasetShareObjects,
+  getShareRequestsToMe,
+  useClient
+} from 'services';
+import { getShareRequestsFromMe } from '../services';
+
+import { ShareBoxListItem } from './ShareBoxListItem';
+import { listDatasets } from '../../Datasets/services';
+import { listAllGroups } from '../../Environments/services';
+//TODO MANAGE DEPENDENCY
+
+const icon = <CheckBoxOutlineBlankIcon fontSize="small" />;
+const checkedIcon = <CheckBoxIcon fontSize="small" />;
+
+export const ShareBoxList = (props) => {
+  const { tab, dataset } = props;
+  const dispatch = useDispatch();
+  const { settings } = useSettings();
+  const [loading, setLoading] = useState(true);
+  const client = useClient();
+  const [items, setItems] = useState(Defaults.pagedResponse);
+  const [filter, setFilter] = useState(Defaults.filter);
+  const [requestGroupOptions, setRequestGroupOptions] = useState([]);
+  const [datasetGroupOptions, setDatasetGroupOptions] = useState([]);
+  const [datasets, setDatasets] = useState([]);
+  const statusOptions = [
+    'Draft',
```

Comment:
Maybe for an enhancement idea - is there a better way to get the shareStatus options rather than have them hardcoded here in the frontend (...could be as a query to the backend that returns the different Enum options)

Also I see we have some hardcoded status strings in `ShareStatus` - not sure if could also extend to there?

Just an idea, but think can be taken up in a later PR

Reply:
Yes, it was in the TODOs comments. I moved it to a constant file in FE. We are starting to have the case that there is shared code between deploy, FE and BE, which leads me to think that it is time to have some globals